### PR TITLE
Demo User & Navbar Refactoring

### PR DESCRIPTION
**Pull request by Duria73**
### What this PR does (required):
- Refactored navbar & added responsive mobile menu button for eventual drawer component. 
- Added demo user function using Login Helper & useAuth with hard-coded user data, routes from landing, login & signup pages. 

### Screenshots / Videos (required):
image
image

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- Hook anchor & drawer to login & signup buttons on mobile, as well as full menu list once pages are completed depending on whether user has completed login or not. 
- Minor style fixes needed on mobile loadouts.

### CODE REVIEW

File: client/src/components/NavBar/useStyles.ts [hunk 1]
```diff
@@ -1,28 +1,45 @@
-import { makeStyles } from '@material-ui/core/styles';
+import { makeStyles, Theme } from '@material-ui/core/styles';
 
-const useStyles = makeStyles((theme) => ({
+const useStyles = makeStyles((theme: Theme) => ({
   root: {
-    backgroundColor: '#FFFFFF',
+    // backgroundColor: '#FFFFFF',
```
Comment (rajivtitus) [inline client/src/components/NavBar/useStyles.ts:5]: You can remove commented styles if we are not currently using them.

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx [hunk 2]
```diff
@@ -8,6 +8,9 @@ import Typography from '@material-ui/core/Typography';
 import useStyles from './useStyles';
 import { CircularProgress } from '@material-ui/core';
 
+import {useAuth} from '../../../context/useAuthContext';
+import login from '../../../helpers/APICalls/login';
+
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:13]: Line 135, the current <Link> component directs to './signup'  please change to redirect to the 'login form'

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx [hunk 3]
```diff
@@ -32,6 +35,15 @@ interface Props {
 
 const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
   const classes = useStyles();
+  const { updateLoginContext } = useAuth()
+
+    const demoUserSubmit = () => {
+      login("demo@hatchways.com", "demo123").then((data) => {
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:41]: I tried logging in as a Demo User with the above credentials, they don't seem to exist.
You can manually create a new user using Postman :)

**Review by rajivtitus — APPROVED** [review]
Good work adding responsiveness to the navbar early on. 
I am approving this PR but please address comments before merging :)

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx (hunk 2 above)
Comment (Duria73) [reply client/src/pages/SignUp/SignUpForm/SignUpForm.tsx:13]: corrected

**Pull request merged**